internal/controller/postgrescluster/pgbackrest.go

```diff
@@ -511,8 +511,9 @@ func (r *Reconciler) setScheduledJobStatus(ctx context.Context,
 // generateRepoHostIntent creates and populates StatefulSet with the PostgresCluster's full intent
 // as needed to create and reconcile a pgBackRest dedicated repository host within the kubernetes
 // cluster.
-func (r *Reconciler) generateRepoHostIntent(postgresCluster *v1beta1.PostgresCluster,
-	repoHostName string) (*appsv1.StatefulSet, error) {
+func (r *Reconciler) generateRepoHostIntent(ctx context.Context,
```

Comment:
⛏️ This calls the API when it did not before. Is it possible to leave API calls to the caller?